### PR TITLE
msg/async: _try_send trim already sent for outcoming_bl more efficient

### PR DESCRIPTION
**Pull request by Yan-waller**
Signed-off-by: Yan Jun <yan.jun8@zte.com.cn>

### CODE REVIEW

File: src/msg/async/AsyncConnection.cc [hunk 1]
```diff
@@ -394,13 +394,8 @@ ssize_t AsyncConnection::_try_send(bool send, bool more)
 
   // trim already sent for outcoming_bl
   if (sent_bytes) {
-    if (sent_bytes < outcoming_bl.length()) {
-      bufferlist bl;
-      outcoming_bl.splice(sent_bytes, outcoming_bl.length()-sent_bytes, &bl);
-      bl.swap(outcoming_bl);
```
Comment (yuyuyu101) [inline src/msg/async/AsyncConnection.cc]: we only need to splice to a nullptr when sent_bytes < outcoming_bl.length()

File: src/msg/async/AsyncConnection.cc [hunk 2]
```diff
@@ -394,13 +394,8 @@ ssize_t AsyncConnection::_try_send(bool send, bool more)
...
-    } else {
-      outcoming_bl.clear();
-    }
+    assert(sent_bytes <= outcoming_bl.length());
+    outcoming_bl.splice(0, sent_bytes);
```
Comment (Yan-waller) [inline src/msg/async/AsyncConnection.cc]: Yes, we want to trim already sent part, from my point of view, It seems to be easier to understand, and achieve the same effect as above (both sent_bytes < outcoming_bl.length() and sent_bytes = outcoming_bl.length()). It seems to be unnecessary to define bl and do swap operation. tks.

Comment (yuyuyu101) [inline src/msg/async/AsyncConnection.cc]: clear is still needed since it will be faster

**Comment by Yan-waller** [comment]
@yuyuyu101 Done.

**Comment by xiexingguo** [comment]
@Yan-waller Need rebase

**Pull request merged**